prover/witness_generator/src/main.rs

```diff
@@ -124,85 +120,112 @@ async fn main() -> anyhow::Result<()> {
         );
     }
 
-    let prometheus_config = if use_push_gateway {
-        PrometheusExporterConfig::push(
-            prometheus_config.gateway_endpoint(),
-            prometheus_config.push_interval(),
-        )
-    } else {
-        PrometheusExporterConfig::pull(prometheus_config.listener_port)
-    };
-    let prometheus_task = prometheus_config.run(stop_receiver.clone());
-
-    let public_object_store_config =
-        PublicObjectStoreConfig::from_env().context("PublicObjectStoreConfig::from_env()")?;
-    let witness_generator_task = match opt.round {
-        AggregationRound::BasicCircuits => {
-            let public_blob_store = match config.shall_save_to_public_bucket {
-                false => None,
-                true => Some(
-                    ObjectStoreFactory::new(public_object_store_config.0)
-                        .create_store()
-                        .await,
-                ),
-            };
-            let generator = BasicWitnessGenerator::new(
-                config,
-                &store_factory,
-                public_blob_store,
-                connection_pool,
-                prover_connection_pool,
-                protocol_versions.clone(),
-            )
-            .await;
-            generator.run(stop_receiver, opt.batch_size)
-        }
-        AggregationRound::LeafAggregation => {
-            let generator = LeafAggregationWitnessGenerator::new(
-                config,
-                &store_factory,
-                prover_connection_pool,
-                protocol_versions.clone(),
-            )
-            .await;
-            generator.run(stop_receiver, opt.batch_size)
-        }
-        AggregationRound::NodeAggregation => {
-            let generator = NodeAggregationWitnessGenerator::new(
-                config,
-                &store_factory,
-                prover_connection_pool,
-                protocol_versions.clone(),
-            )
-            .await;
-            generator.run(stop_receiver, opt.batch_size)
-        }
-        AggregationRound::Scheduler => {
-            let generator = SchedulerWitnessGenerator::new(
-                config,
-                &store_factory,
-                prover_connection_pool,
-                protocol_versions,
-            )
-            .await;
-            generator.run(stop_receiver, opt.batch_size)
+    let rounds = match (opt.round, opt.all_rounds) {
+        (Some(round), false) => vec![round],
+        (None, true) => vec![
+            AggregationRound::BasicCircuits,
+            AggregationRound::LeafAggregation,
+            AggregationRound::NodeAggregation,
+            AggregationRound::Scheduler,
+        ],
+        _ => {
+            return Err(anyhow!(
+                "Expected --all_rounds flag with no --round flag present"
```

Comment:
nit: this is not fully correct - as this condition also triggers, when you specify both --all_rounds and --round=X flag.